### DISCUSSION TITLE
Equivalent to "GuildMemberRoleAddEvent" for offline users

### DISCUSSION THREAD

**Discussion by sloenthran** [discord-jda/JDA#1588] [Questions and Help] [answered]
Hi.

I have a problem. My bot is supposed to post an entry in the database automatically after assigning a rank, but it does so only when someone is online. So what is the counterpart of this event for offline people? I would be very grateful for your help :)

**Comment by MinnDevelopment** [comment, marked as answer]
This event has nothing to do with the online state of a member. The event requires members to be cached in order to figure out which roles were added. You can use [GuildMemberUpdateEvent](https://ci.dv8tion.net/job/JDA/javadoc/net/dv8tion/jda/api/events/guild/member/update/GenericGuildMemberUpdateEvent.html) instead. Please read the documentation for [GuildMemberRoleAddEvent](https://ci.dv8tion.net/job/JDA/javadoc/net/dv8tion/jda/api/events/guild/member/GuildMemberRoleAddEvent.html) which directly explains the requirements for its trigger.
